### PR TITLE
git-credential-gopass: init at 1.12.0

### PR DESCRIPTION
**Pull request by benneti**
###### Motivation for this change
git-credential-gopass was split from gopass and now needs it's own little package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/gopass/git-credential.nix
Comment (worldofpeace) [inline pkgs/tools/security/gopass/git-credential.nix:38]:
```suggestion
    homepage = "https://github.com/gopasspw/git-credential-gopass";
```

the website doesn't show anything about this particular program. I think the github page should suffice for now.

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix:1419]: I think this should be under `gitAndTools`.

File: maintainers/maintainer-list.nix
Comment (worldofpeace) [inline maintainers/maintainer-list.nix:10776]: if you could do this in a separate commit that would be great. we usually do this because reverts or backports are better that way (and a bit nicer history).

**Comment by worldofpeace** [comment]
Thanks a lot for this! :sparkles: I legitimately thought it was just gone.

File: pkgs/top-level/all-packages.nix
Comment (benneti) [reply pkgs/top-level/all-packages.nix:1419]: just the call or should I also move the .nix file?
I put it there because it only makes sense for people using gopass anyway

File: pkgs/tools/security/gopass/git-credential.nix
Comment (benneti) [reply pkgs/tools/security/gopass/git-credential.nix:38]: Very reasonable

File: pkgs/top-level/all-packages.nix
Comment (worldofpeace) [reply pkgs/top-level/all-packages.nix:1419]: @benneti Actually, nvm we killed it about a month ago https://github.com/NixOS/nixpkgs/pull/109396

File: pkgs/tools/security/gopass/git-credential.nix
Comment (benneti) [reply pkgs/tools/security/gopass/git-credential.nix]: done

File: maintainers/maintainer-list.nix
Comment (benneti) [reply maintainers/maintainer-list.nix:10776]: done

**Comment by benneti** [comment]
> Thanks a lot for this! sparkles I legitimately thought it was just gone.

Me too for a very sad, brief moment :sweat_smile: 
I hope checks out now.

**Comment by worldofpeace** [comment]
> > Thanks a lot for this! sparkles I legitimately thought it was just gone.
> 
> Me too for a very sad, brief moment sweat_smile
> I hope checks out now.

Currently building, but it all checks out. Btw, just a quick tip for contributing to nixpkgs (if u do in the future) is we prefer PRs to be on unique branches other than `master`. So u can just `git checkout upstream/master` `git checkout -b some-branch-name`.

**Review by worldofpeace — APPROVED** [review]
LGTM

File: pkgs/tools/security/gopass/git-credential.nix
Comment (worldofpeace) [inline pkgs/tools/security/gopass/git-credential.nix:24]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/tools/security/gopass/git-credential.nix
Comment (worldofpeace) [inline pkgs/tools/security/gopass/git-credential.nix:5]:
```suggestion
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Oops, I take that back. I don't see this doing anything in the output

**Comment by benneti** [comment]
Ah sorry forgot that when copying from gopass/default.nix...

**Comment by worldofpeace** [comment]
Thanks for contributing :sparkles:

**Comment by benneti** [comment]
Thanks for the fast feedback! :tada:

**Comment by SuperSandro2000** [comment]
The commit history should have been cleaned up before this was merged.

File: pkgs/tools/security/gopass/git-credential.nix
Comment (SuperSandro2000) [inline pkgs/tools/security/gopass/git-credential.nix:40]: platforms is set by buildGoModule and shouldn't be set here.

**Comment by worldofpeace** [comment]
> The commit history should have been cleaned up before this was merged.

It was squash merged.

File: pkgs/tools/security/gopass/git-credential.nix
Comment (benneti) [reply pkgs/tools/security/gopass/git-credential.nix:40]: should I do a new PR for this?

Comment (worldofpeace) [reply pkgs/tools/security/gopass/git-credential.nix:40]: If you wish, but no one can *make you* if that makes sense.

Comment (benneti) [reply pkgs/tools/security/gopass/git-credential.nix:40]: That's clear.

**Comment by SuperSandro2000** [comment]
> It was squash merged.

So the adding the maintainer is not an extra commit? Also The maintainer is not sorted alphabetically. Going to fix that in a fix collection PR.

**Pull request merged**
